Feat/vasttrafik update (#40566)
home-assistant/home-assistant.github.io: source/_integrations/vasttrafik.markdown

ha_domain: vasttrafik
ha_platforms:
  - sensor
ha_integration_type: service
ha_config_flow: true
related:
  - docs: /docs/configuration/
    title: Configuration file
---

The `vasttrafik` {% term integration %} will provide you traveling details for the larger Göteborg area in Sweden from the [Västtrafik](https://vasttrafik.se/) public transportation service.

## Prerequisites

- You must create an application [via the vasttrafik developer page](https://developer.vasttrafik.se/applications) to obtain a `key` and a `secret`. 
- Subscribe to the API by selecting **Prenumerera på nytt API**, and selecting the API **Planera Resa v4**.

{% include integrations/config_flow.md %}

### Adding departure boards

Once the main integration is configured, you can add departure boards:

1. Go to {% my integrations title="**Settings** > **Devices & services** > **Västtrafik**" domain="vasttrafik" %}
2. Select **Add departure board**
3. Search for your departure station (for example, "Central", "Musikvägen")

> 4. Select the station from the results

Reviewer: ```suggestion
1. Go to {% my integrations title="**Settings** > **Devices & services** > **Västtrafik**" domain="vasttrafik" %}.
2. Select **Add departure board**.
3. Search for your departure station (for example, "Central", "Musikvägen").
4. Select the station from the results.
```

5. Configure your departure board:
   - **Name**: Optional custom name for the sensor
   - **Destination**: Optional filter for specific destination (for example, "Järntorget")
   - **Lines**: Optional comma-separated list of line numbers (for example, "1, 2, 7")
   - **Tracks/Platforms**: Optional comma-separated list (for example, "A, B, 1")
   - **Delay**: Minutes to add to current time for departure lookup (default: 0)

## Troubleshooting station selection

The departure board configuration includes an interactive station search. If the wrong station is being selected during setup:

1. Try using the full station name (for example, "Musikvägen, Göteborg")
2. Use more specific search terms to narrow down results

> 3. Try alternative names or spellings of the station

Reviewer: ```suggestion
1. Try using the full station name (for example, "Musikvägen, Göteborg").
2. Use more specific search terms to narrow down results.
3. Try alternative names or spellings of the station.
```

If problems persist, you can find the exact station ID for troubleshooting:

1. Login to the Västtrafik API and go to ["Applikationer"](https://developer.vasttrafik.se/applications)

> 2. Select **\* Generera accesstoken**, then **Kopiera**

Reviewer: ```suggestion
1. Login to the Västtrafik API and go to ["Applikationer"](https://developer.vasttrafik.se/applications).
2. Select **\* Generera accesstoken**, then **Kopiera**.
```

3. Execute this `curl` command, replacing `<ACCESS_TOKEN>` and `<STATION_NAME>`:

   ```shell
   curl -H "Authorization: Bearer <ACCESS_TOKEN>" "https://ext-api.vasttrafik.se/pr/v4/locations/by-text?q=<STATION_NAME>"
   ```

4. In the output, find the "gid" for your desired stop. If the UI search doesn't work with this station name, please report the issue to the integration maintainers.

## Migration from YAML configuration

If you previously configured Västtrafik in your `configuration.yaml` file, your existing configuration will be automatically imported when you upgrade to a version that includes this feature.

After the automatic import:

1. A repair notification will be created in **Settings** > **System** > **Repairs**
2. Your existing departure boards will be migrated as sub-entries under the main Västtrafik integration
3. You can safely remove the YAML configuration from your `configuration.yaml` file after verifying the import was successful

> 4. Restart Home Assistant to complete the migration

Reviewer: ```suggestion
1. A repair notification will be created in **Settings** > **System** > **Repairs**.
2. Your existing departure boards will be migrated as sub-entries under the main Västtrafik integration.
3. You can safely remove the YAML configuration from your `configuration.yaml` file after verifying the import was successful.
4. Restart Home Assistant to complete the migration.
```

> 

Reviewer: ```suggestion

## Prerequisites

```

> You must create an application [here](https://developer.vasttrafik.se/applications) to obtain a `key` and a `secret`. Make sure to also subscribe to the API by clicking on `Prenumerera på nytt API`, and selecting the API `Planera Resa v4`.

Reviewer: ```suggestion
- You must create an application [via the vasttrafik developer page](https://developer.vasttrafik.se/applications) to obtain a `key` and a `secret`. 
- Subscribe to the API by selecting **Prenumerera på nytt API**, and selecting the API **Planera Resa v4**.
```
We use [bold to markup UI strings](https://developers.home-assistant.io/docs/documenting/general-style-guide)

## Configuration

Västtrafik is configured via the UI. Go to **Settings** → **Devices & Services** → **Add Integration** and search for "Västtrafik".

> You will need your API credentials obtained in the prerequisites section above.

Reviewer: ```suggestion

{% include integrations/config_flow.md %}
```
This section is covered by this reuse element. See https://developers.home-assistant.io/docs/documenting/integration-docs-examples#configuration, and https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown#prerequisites

### Initial Setup

1. Add the Västtrafik integration from **Settings** → **Devices & Services**
2. Enter your API key and secret when prompted

> 3. The integration will be added as a service

Reviewer: ```suggestion
Use the configuration_basic element instead.
```

> ### Adding Departure Boards

Reviewer: Please [check the style guide]() and fix throughout the docs:
- [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization)
- We prefer [select over click](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click)
- use for example instead of e.g.,

Author: This should be addressed in the latest commit

Once the main integration is configured, you can add departure boards:

> 1. Go to **Settings** → **Devices & Services** → **Västtrafik**

Reviewer: ```suggestion
1. Go to {% my integrations title="**Settings** > **Devices & services** > **Västtrafik**" domain="vasttrafik" %}
```
